tests/thread_float/main.c

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (C) 2017 OTA keys S.A.
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup tests
+ * @{
+ *
+ * @file
+ * @brief Thread test application
+ *
+ * @author Vincent Dupont <vincent@otakeys.com>
+ *
+ * @}
+ */
+
+#include <stdio.h>
+
+#include "thread.h"
+#include "msg.h"
+#include "xtimer.h"
+#include "timex.h"
+
+static char t1_stack[THREAD_STACKSIZE_MAIN];
+static char t2_stack[THREAD_STACKSIZE_MAIN];
+static char t3_stack[THREAD_STACKSIZE_MAIN];
+
+static kernel_pid_t p1, p2, p3;
+
+static xtimer_t timer;
+#define OFFSET (10 * XTIMER_BACKOFF)
+
+static mutex_t lock = MUTEX_INIT;
+
+static void timer_cb(void *arg)
+{
+    (void) arg;
+
+    xtimer_set(&timer, OFFSET);
+    sched_switch(THREAD_PRIORITY_MAIN - 2);
+}
+
+static void *thread1(void *arg)
+{
+    (void) arg;
+
+    float f, init;
+
+    printf("THREAD %" PRIkernel_pid " start\n", thread_getpid());
+
+    init = 1.0 * thread_getpid();
+    f = init;
+
+    while (1) {
+        for (int i = 0; i < 10000; i++) {
+            f = f + 1.0 / f;
+        }
+        mutex_lock(&lock);
+        printf("T(%" PRIkernel_pid "): %f\n", thread_getpid(), (double)f);
+        mutex_unlock(&lock);
+        init += 1.0;
+        f = init;
+    }
+    return NULL;
+}
+
+static void *thread2(void *arg)
+{
+    (void) arg;
+
+    float f, init;
+
+    printf("THREAD %" PRIkernel_pid " start\n", thread_getpid());
+
+    init = 1.0 * thread_getpid();
+    f = init;
+
+    while (1) {
+        for (int i = 0; i < 100000; i++) {
```

Comment:
Seems murdock doesn't like this too much, maybe specify uint32_t or limit it to 32767.  If you do that I will test then hopefully merge now that I am a maintainer ;)

Reply:
I think you hit the wrong for loop, do the one on line 83.

Reply:
Sorry, I had fixed the other one, this one should be fixed now as well

Reply:
On the AVR (among others) an `int` is only 16 bits, so this comparison is always true for those architectures.

Reply:
@bergzand I switch to `unsigned long`